components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/service/impl/ActionManagementServiceImpl.java

```diff
@@ -38,26 +43,22 @@
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 /**
  * Action management service.
  */
 public class ActionManagementServiceImpl implements ActionManagementService {
 
     private static final Log LOG = LogFactory.getLog(ActionManagementServiceImpl.class);
-    private static final ActionManagementService INSTANCE = new ActionManagementServiceImpl();
-    private static final CacheBackedActionMgtDAO CACHE_BACKED_DAO =
-            new CacheBackedActionMgtDAO(new ActionManagementDAOImpl());
     private static final ActionValidator ACTION_VALIDATOR = new ActionValidator();
     private static final ActionManagementAuditLogger auditLogger = new ActionManagementAuditLogger();
-    private static final ActionSecretProcessor ACTION_SECRET_PROCESSOR = new ActionSecretProcessor();
 
-    private ActionManagementServiceImpl() {
-    }
+    private final ActionManagementDAOFacade daoFacade;
 
-    public static ActionManagementService getInstance() {
+    public ActionManagementServiceImpl(ActionManagementDAO actionManagementDAO) {
```

Comment:
IMO, we should not pass a DAO for the service layer

Reply:
Fixed